src/librbd/image/OpenRequest.cc

```diff
@@ -30,11 +30,11 @@ using util::create_context_callback;
 using util::create_rados_ack_callback;
 
 template <typename I>
-OpenRequest<I>::OpenRequest(I *image_ctx, bool skip_open_parent,
-                            Context *on_finish)
-  : m_image_ctx(image_ctx), m_skip_open_parent_image(skip_open_parent),
-    m_on_finish(on_finish), m_error_result(0),
-    m_last_metadata_key(ImageCtx::METADATA_CONF_PREFIX) {
+OpenRequest<I>::OpenRequest(I *image_ctx, bool skip_get_name,
```

Comment:
I'm also in favor for the second option. Are you suggesting that I drop the `skip_get_name` bool and first try to get the image name from the rbd_directory and if it does not exist, try to get the name from the rbd_trash?